txpool/txpool.go

```diff
@@ -133,10 +140,57 @@ func (p *txPool) checkTx(trx *tx.Tx) error {
 	return nil
 }
 
-func (p *txPool) RemoveTx(id tx.ID) {
+func (p *txPool) HandleCommittedBlock(blk *block.Block) error {
 	p.lk.Lock()
 	defer p.lk.Unlock()
 
+	for _, trx := range blk.Transactions() {
+		p.removeTx(trx.ID())
+
+		p.handleIncreaseConsumption(trx)
+	}
+
+	return p.handleDecreaseConsumption(blk.Height())
+}
+
+func (p *txPool) handleIncreaseConsumption(trx *tx.Tx) {
+	if trx.IsTransferTx() || trx.IsBondTx() || trx.IsWithdrawTx() {
+		signer := trx.Payload().Signer()
+
+		// retrieve existing consumption or start with 0
+		p.consumptionMap[signer] = p.consumptionMap[signer] + uint32(trx.SerializeSize())
+	}
+}
+
+func (p *txPool) handleDecreaseConsumption(height uint32) error {
+	if height <= p.config.ConsumptionBlocks {
+		return nil
+	}
+
+	oldConsumptionHeight := height - p.config.ConsumptionBlocks
+	committedBlock, err := p.strReader.Block(oldConsumptionHeight)
+	if err != nil {
+		return err
+	}
+
+	blk, err := committedBlock.ToBlock()
+	if err != nil {
+		return err
+	}
+
+	for _, trx := range blk.Transactions() {
+		if trx.IsTransferTx() || trx.IsBondTx() || trx.IsWithdrawTx() {
+			signer := trx.Payload().Signer()
+			if v, ok := p.consumptionMap[signer]; ok {
+				p.consumptionMap[signer] = v - uint32(trx.SerializeSize())
```

Comment:
@b00f I don't know on test `TestCalculatingConsumption` why value `p.consumptionMap` for specific signer don't update.